src/test/java/core/basesyntax/service/impl/WorkerWithFruitsTransactionsTest.java

```diff
@@ -0,0 +1,135 @@
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.CantGetFruitException;
+import core.basesyntax.exceptions.CantPutFruitException;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.WorkerWithTransactions;
+import core.basesyntax.strategy.OperationStrategyImpl;
+import core.basesyntax.strategy.handler.BalanceOperationHandler;
+import core.basesyntax.strategy.handler.OperationHandler;
+import core.basesyntax.strategy.handler.PurchaseOperationHandler;
+import core.basesyntax.strategy.handler.ReturnOperationHandler;
+import core.basesyntax.strategy.handler.SupplyOperationHandler;
+import java.util.Map;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class WorkerWithFruitsTransactionsTest {
+    public static final Operation BALANCE_OPERATION = Operation.BALANCE;
+    public static final Operation SUPPLY_OPERATION = Operation.SUPPLY;
+    public static final Operation PURCHASE_OPERATION = Operation.PURCHASE;
+    public static final Operation RETURN_OPERATION = Operation.RETURN;
+    public static final String APPLE = "apple";
+    public static final String BANANA = "banana";
+    public static final Integer QUANTITY_100 = 100;
+    public static final Integer QUANTITY_50 = 50;
+    public static final Integer QUANTITY_30 = 30;
+    public static final Integer QUANTITY_130 = 130;
+    private final Map<Operation, OperationHandler> operation = Map.of(
+            Operation.BALANCE, new BalanceOperationHandler(),
+            Operation.RETURN, new ReturnOperationHandler(),
+            Operation.PURCHASE, new PurchaseOperationHandler(),
+            Operation.SUPPLY, new SupplyOperationHandler());
+    private final WorkerWithTransactions workerWithTransactions
+            = new WorkerWithFruitsTransactions(new OperationStrategyImpl(operation));
+
+    @BeforeEach
+    void setUp() {
+        Storage.getStorage().clear();
+    }
+
+    @Test
+    void completeTransaction_setBalance_Ok() {
+        Transaction transaction = new FruitTransaction(BALANCE_OPERATION, APPLE, QUANTITY_100);
+        workerWithTransactions.completeTransaction(transaction);
+        Integer actual = Storage.getStorage().get(APPLE);
+        Assertions.assertEquals(Integer.valueOf(100), actual);
+    }
+
+    @Test
+    void completeTransaction_setBalanceDifferentFruit_Ok() {
+        createAndCompleteTransactions(BALANCE_OPERATION, APPLE, QUANTITY_100);
+        createAndCompleteTransactions(BALANCE_OPERATION, BANANA, QUANTITY_50);
+        boolean isAppleExistInStorage = Storage.getStorage().containsKey(APPLE);
+        boolean isBananaExistInStorage = Storage.getStorage().containsKey(BANANA);
+        Integer actual = Storage.getStorage().get(APPLE);
+        Assertions.assertTrue(isAppleExistInStorage && isBananaExistInStorage);
```

Comment:
The variable 'actual' isn't being used
```suggestion
        Assertions.assertTrue(Storage.getStorage().containsKey(APPLE));
        Assertions.assertTrue(Storage.getStorage().containsKey(BANANA));
```

Reply:
> The variable 'actual' isn't being used

+